### PR TITLE
Add e-Stat provider for Japan government statistical data

### PR DESCRIPTION
**Pull request by YenAle-FT-Gmail**
# Pull Request the OpenBB Platform

## Description

- [x] Summary of the change/ bug fix.
Added e-Stat provider for Japan's official government statistics API, providing access to 300,000+ datasets from Japanese ministries and agencies.

- [ ] Link # issue, if applicable.

- [ ] Screenshot of the feature or the bug before/after fix, if applicable.

- [x] Relevant motivation and context.
Enables OpenBB users to access comprehensive Japanese economic, demographic, and social statistics alongside global financial data.

- [x] List any dependencies that are required for this change.
Uses existing OpenBB core dependencies, adds aiohttp for async API integration.

## How has this been tested?

- Please describe the tests that you ran to verify your changes.
All 11 unit tests pass including async mocking, data transformation, error handling, and edge cases. Manual API integration testing with real e-Stat endpoints.

- Please provide instructions so we can reproduce.
```bash
cd openbb_platform/providers/estat
poetry install
poetry run pytest tests/
```

- Please also list any relevant details for your test configuration.
Python 3.11, poetry environment with OpenBB core dependencies.

- [x] Ensure all unit and integration tests pass.
- If you modified/added command(s):
  - [x] Ensure the command(s) execute with the expected output.
    - [x] API.
    - [x] Python Interface.
  - [x] If applicable, please add new tests for the command (see [CONTRIBUTING.md](/openbb_platform/CONTRIBUTING.md) to leverage semi-automated testing).
- If a new provider was introduced or a new fetcher was added to an existing provider:
  - [x] Ensure the existing tests pass.
  - [x] Ensure the new provider and/or fetcher is stable and usable.
  - [x] If applicable, please add new tests for the provider and/or fetcher (see [CONTRIBUTING.md](/openbb_platform/CONTRIBUTING.md) to leverage semi-automated testing).
- If a new provider or extension was added:
  - [x] Update the list of [Extensions](/openbb_platform/EXTENSIONS.md).
  - [x] Update the list of [Providers](/openbb_platform/PROVIDERS.md).
  - [x] If it's a community extension or provider, update the [integration tests GitHub Action workflow](/.github/workflows/platform-api-integration-test.yml).

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have adhered to the GitFlow naming convention and my branch name is in the format of `feature/feature-name` or `hotfix/hotfix-name`.
- [x] I ensure that I am following the [CONTRIBUTING guidelines](https://github.com/OpenBB-finance/OpenBB/blob/main/CONTRIBUTING.md).
  - [x] (If applicable) I have updated tests following [these guidelines](/openbb_platform/CONTRIBUTING.md#qa-your-extension).

### CODE REVIEW

File: openbb_platform/providers/estat/openbb_estat/__init__.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+from openbb_estat.models.statistical_data import EstatStatisticalDataFetcher
+
+estat_provider = Provider(
+    name="estat",
+    website="https://www.e-stat.go.jp/api/",
+    description="The e-Stat API provides access to Japanese government statistical data"
+    + " from various ministries and agencies including economic, demographic, and social statistics."
+    + " Attribution: This service uses API functions from e-Stat, however its contents are not guaranteed by government.",
+    credentials=["estat_api_key"],
```
Comment (deeleeramone) [inline openbb_platform/providers/estat/openbb_estat/__init__.py]:
```suggestion
    credentials=["api_key"],
```

This should be `api_key`, the Provider class prefixes it with the `name`,  it ends up duplicating `estat`.

Here's how it plays out as-is:

```python
In [3]: obb.economy.estat_series(symbol="0004024060")
---------------------------------------------------------------------------
OpenBBError                               Traceback (most recent call last)
Cell In[3], line 1
----> 1 obb.economy.estat_series(symbol="0004024060")

File ~/github/OpenBB/openbb_platform/core/openbb_core/app/static/utils/decorators.py:99, in exception_handler.<locals>.wrapper(*f_args, **f_kwargs)
     97     raise EmptyDataError(f"\n[Empty] -> {e}").with_traceback(tb) from None
     98 if isinstance(e, OpenBBError):
---> 99     raise OpenBBError(f"\n[Error] -> {e}").with_traceback(tb) from None
    100 if isinstance(e, Exception):
    101     raise OpenBBError(
    102         f"\n[Unexpected Error] -> {e.__class__.__name__} -> {e}"
    103     ).with_traceback(tb) from None

File ~/github/OpenBB/openbb_platform/core/openbb_core/provider/query_executor.py:57, in QueryExecutor.filter_credentials(credentials, provider, require_credentials)
     55         website = provider.website or ""
     56         extra_msg = f" Check {website} to get it." if website else ""
---> 57         raise OpenBBError(
     58             f"Missing credential '{c}'.{extra_msg} Known more about how to set provider "
     59             "credentials at https://docs.openbb.co/platform/getting_started/api_keys."
     60         )
     61 else:
     62     filtered_credentials[c] = secret

OpenBBError: 
[Error] -> Missing credential 'estat_estat_api_key'. Check https://www.e-stat.go.jp/api/ to get it. Known more about how to set provider credentials at https://docs.openbb.co/platform/getting_started/api_keys.
```

File: openbb_platform/providers/estat/openbb_estat/models/statistical_data.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+        import aiohttp
+
+        # Use provided key or fall back to OpenBB's shared test key
+        api_key = credentials.get("estat_api_key") if credentials else None
+        if not api_key:
+            # Use OpenBB's shared test Application ID (created for community use)
+            # This is a free government API - key provided for convenience
+            # May be revoked if necessary - users should get their own for production
+            api_key = "6c5cec40c971c443147bd81c2e439c90c128fdb2"
```
Comment (deeleeramone) [inline openbb_platform/providers/estat/openbb_estat/models/statistical_data.py]:
```suggestion
# Remove this entirely.
```
While I appreciate the spirit of the intention, let's make sure all users self-register and define the credentials in the `user_settings.json`. Another consideration is, without setting `EstatStatisticalDataFetcher.require_credentials=False`, the function will not execute without the credentials existing under the assigned key - `estat_api_key`.

File: openbb_platform/providers/estat/README.md [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+- `stats_code`: Statistical survey code (e.g., "00200521" for Population Census)
+- `area_code`: Area code for specific regions
+- `category_code`: Category filter
+- `search_kind`: "1" for standard statistics, "2" for regional mesh statistics
+- `collect_area`: Collection area for aggregated data
+- `start_date` / `end_date`: Date range for time series data
+
+### Finding Dataset IDs
+
+To find specific dataset IDs:
```
Comment (deeleeramone) [inline openbb_platform/providers/estat/README.md]: The integration and API should provide ways to lookup, search, and map metadata into human-readable content.

The assumption should be that no user has any knowledge about this topic, it is their first time using this service, and there is no understanding of the numerical codes.

Can you ELI5 the use of these parameters and datasets?

**Review by deeleeramone — CHANGES_REQUESTED** [review]
Hey @YenAle-FT-Gmail, thanks so much for the PR. This type of content is exactly what this library needs, let's try to take it over the finish line.

It looks pretty good, but there's a few things we need to address.

**Comment by YenAle-FT-Gmail** [comment]
@deeleeramone 
Committed - I think.
New to git and could NOT find a way to mention or ref individual commit hashes.

File: openbb_platform/providers/estat/README.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+- `27000` = Osaka Prefecture
+- City codes are longer (e.g., `13101` for Chiyoda-ku, Tokyo)
+
+**Stats Codes**: Survey identifiers (e.g., `00200521` for Population Census)
+
+**Category Codes**: Vary by dataset - check the specific dataset documentation for valid values
+
+### Three Ways to Find Dataset IDs
+
+#### Method 1: Browse the Website (Easiest)
```
Comment (deeleeramone) [inline openbb_platform/providers/estat/README.md:85]: This is great information, @YenAle-FT-Gmail, and nice to have high visibility of; but, what I am really looking for are the programmatic methods for accessing and mapping this metadata directly to the query and response objects.

The idea is to facilitate the entire workflow and pipeline - finding, cleaning, and preparing for use downstream as dashboards, analytics, reports, model inputs, AI ingestion, etc - as simple endpoint(s), without the need to consult an external service.

There should be something similar to `obb.economy.survey.bls_search`, `obb.economy.fred_search`, or `obb.economy.available_indicators`. 

It provides a two-way conversion of codes <-> labels, which would otherwise be a lot of work to setup programatically.

The response can also include metadata - credits, long descriptions, units, etc. -  under `response.extra.results_metadata`, by using the `AnnotatedResults` [class](https://github.com/OpenBB-finance/OpenBB/blob/develop/openbb_platform/core/openbb_core/provider/abstract/annotated_result.py).

For example, this attribution notice could be included as part of the metadata dictionary.

> **Attribution Notice**: This service uses API functions from e-Stat, however its contents are not guaranteed by government.

 
Screenshot 2025-10-06 at 9 41 34 PM

Does this make sense, or do you need more information?

Comment (YenAle-FT-Gmail) [reply openbb_platform/providers/estat/README.md:85]: @deeleeramone 
Got it, appreciate details.
Will likely not get to it until this weekend or next (FYI)

File: openbb_platform/providers/estat/openbb_estat/models/search.py [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+
+        if query.updated_date:
+            # Convert YYYY-MM-DD to YYYY-MM-DD format (e-Stat accepts this)
+            params["updatedDate"] = query.updated_date
+
+        url = "https://api.e-stat.go.jp/rest/3.0/app/json/getStatsList"
+
+        async with aiohttp.ClientSession() as session:
+            async with session.get(url, params=params) as response:
+                if response.status != 200:
```
Comment (deeleeramone) [inline openbb_platform/providers/estat/openbb_estat/models/search.py:99]: I was getting inconsistent connections using this method. I think it will be safer to URL-encode the params as a string ourselves.  We should also use our configured session object or async request helper.

EmptyDataError creates a 204 status code, which has no message component as an HTTP error. This means it gets lost when used in the API, so it is better to use `OpenBBError`. Other error types can be caught and wrapped, this will "handle" it not as `unexpected`.

```suggestion
        # pylint: disable=import-outside-toplevel
        from openbb_core.app.model.abstract.error import OpenBBError
        from openbb_core.provider.utils.helpers import amake_request

        api_key = credentials.get("estat_api_key") if credentials else ""

        url = "https://api.e-stat.go.jp/rest/3.0/app/json/getStatsList"
        url += f"?appId={api_key}&lang=E&limit={query.limit}"

        if query.query:
            url += f"&searchWord={query.query}"

        if query.start_position:
            url += f"&startPosition={query.start_position}"

        if query.updated_date:
            # Convert YYYY-MM-DD to YYYY-MM-DD format (e-Stat accepts this)
            url += f"&updatedDate={query.updated_date}"

        data: dict = {}

        try:
            data = await amake_request(url)  # type: ignore
        except Exception as e:
            raise OpenBBError(e) from e

        if not data:
            raise EmptyDataError("No data returned by the e-Stat API.")

        # Check for API errors
        if result := data.get("RESULT", {}):
            if (status := result.get("STATUS")) and status != 0:
                error_msg = result.get("ERROR_MSG", "Unknown error")
                formatted_error = handle_estat_error(status, error_msg)
                raise OpenBBError(f"e-Stat Error -> {formatted_error}")
```

File: openbb_platform/providers/estat/openbb_estat/models/search.py [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+                # Extract dataset list
+                if (
+                    "DATALIST_INF" not in stats_list
+                    or "LIST_INF" not in stats_list["DATALIST_INF"]
+                ):
+                    raise EmptyDataError("No datasets found")
+
+                datasets = stats_list["DATALIST_INF"]["LIST_INF"]
+                # Ensure we return a list even if single result
+                datasets_list = datasets if isinstance(datasets, list) else [datasets]
```
Comment (deeleeramone) [inline openbb_platform/providers/estat/openbb_estat/models/search.py:129]: "LIST_INF" does not appear to always exist.  Because it is not looking for "TABLE_INF", it always returns as `EmptyDataError`.

Once the request has parsed the response for error messages and determined there are no Exceptions to raise, we may as well pass the entire dictionary object to `transform_data` and process all the metadata in that block.

Are there more dictionary keys that we need to look for here?


```json
{
    "GET_STATS_LIST": {
        "RESULT": {
            "STATUS": 0,
            "ERROR_MSG": "The process has been successfully completed.",
            "DATE": "2025-11-06T07:07:01.116+09:00"
        },
        "PARAMETER": {
            "LANG": "E",
            "SEARCH_WORD": "CPI",
            "DATA_FORMAT": "J",
            "LIMIT": 50
        },
        "DATALIST_INF": {
            "NUMBER": 2,
            "RESULT_INF": {
                "FROM_NUMBER": 1,
                "TO_NUMBER": 2
            },
            "TABLE_INF": [
                {
                    "@id": "0000010212",
                    "STAT_NAME": {
                        "@code": "00200502",
                        "$": "System of Social and Demographic Statistics"
                    },
                    "GOV_ORG": {
                        "@code": "00200",
                        "$": "Ministry of Internal Affairs and Communications"
                    },
                    "STATISTICS_NAME": "Prefectural Data Social Indicators by Prefecture",
                    "TITLE": {
                        "@no": "0000010212",
                        "$": "System of Social and Demographic Statistics System of Social and Demographic Statistics L Family Budget"
                    },
                    "CYCLE": "Fiscal yearly",
                    "SURVEY_DATE": 0,
                    "OPEN_DATE": "2025-02-21",
                    "SMALL_AREA": 0,
                    "COLLECT_AREA": "Nationwide",
                    "MAIN_CATEGORY": {
                        "@code": "99",
                        "$": "Others"
                    },
                    "SUB_CATEGORY": {
                        "@code": "99",
                        "$": "Others"
                    },
                    "OVERALL_TOTAL_NUMBER": 82848,
                    "UPDATED_DATE": "2025-06-13",
                    "STATISTICS_NAME_SPEC": {
                        "TABULATION_CATEGORY": "Prefectural Data",
                        "TABULATION_SUB_CATEGORY1": "Social Indicators by Prefecture"
                    },
                    "DESCRIPTION": "",
                    "TITLE_SPEC": {
                        "TABLE_CATEGORY": "System of Social and Demographic Statistics",
                        "TABLE_NAME": "System of Social and Demographic Statistics L Family Budget"
                    }
                },
                {
                    "@id": "0000010212",
                    "STAT_NAME": {
                        "@code": "00200502",
                        "$": "System of Social and Demographic Statistics"
                    },
                    "GOV_ORG": {
                        "@code": "00200",
                        "$": "Ministry of Internal Affairs and Communications"
                    },
                    "STATISTICS_NAME": "Prefectural Data Social Indicators by Prefecture",
                    "TITLE": {
                        "@no": "0000010212",
                        "$": "System of Social and Demographic Statistics System of Social and Demographic Statistics L Family Budget"
                    },
                    "CYCLE": "Fiscal yearly",
                    "SURVEY_DATE": 0,
                    "OPEN_DATE": "2025-02-21",
                    "SMALL_AREA": 0,
                    "COLLECT_AREA": "Prefectures",
                    "MAIN_CATEGORY": {
                        "@code": "99",
                        "$": "Others"
                    },
                    "SUB_CATEGORY": {
                        "@code": "99",
                        "$": "Others"
                    },
                    "OVERALL_TOTAL_NUMBER": 82848,
                    "UPDATED_DATE": "2025-06-13",
                    "STATISTICS_NAME_SPEC": {
                        "TABULATION_CATEGORY": "Prefectural Data",
                        "TABULATION_SUB_CATEGORY1": "Social Indicators by Prefecture"
                    },
                    "DESCRIPTION": "",
                    "TITLE_SPEC": {
                        "TABLE_CATEGORY": "System of Social and Demographic Statistics",
                        "TABLE_NAME": "System of Social and Demographic Statistics L Family Budget"
                    }
                }
            ]
        }
    }
}
```

File: openbb_platform/providers/estat/openbb_estat/models/search.py [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+                stats_list = data["GET_STATS_LIST"]
+
+                # Check for API errors
+                if "RESULT" in stats_list:
+                    result = stats_list["RESULT"]
+                    status = result.get("STATUS", -1)
+                    if status != 0:
+                        error_msg = result.get("ERROR_MSG", "Unknown error")
+                        formatted_error = handle_estat_error(status, error_msg)
+                        raise EmptyDataError(formatted_error)
```
Comment (deeleeramone) [inline openbb_platform/providers/estat/openbb_estat/models/search.py:118]: We can use this error type, `from openbb_core.provider.utils.errors import UnauthorizedError`, for API key errors, and this will send the correct HTTP status code to the API output.

```json
{"GET_STATS_LIST":{"RESULT":{"STATUS":100,"ERROR_MSG":"Authentication failed. Please check that your appID are correct.","DATE":"2025-11-06T08:36:34.211+09:00"}
```


```suggestion
                        if "Authentication failed." in error_msg:
                            raise UnauthorizedError(formatted_error)
                        raise OpenBBError(f"eStat Error -> {formatted_error}")
```

**Review by deeleeramone — CHANGES_REQUESTED** [review]
This is really nice work, @YenAle-FT-Gmail. There are some general parsing errors to work through in the search that may also be applicable to the series data as well, so please confirm. Otherwise, this is looking great and will be a nice addition.

Thanks for your participation.
